### DISCUSSION TITLE
How to install?

### DISCUSSION THREAD

**Discussion by thany** [openai/whisper#1939] [General]
Can a proper good step-by-step guide please be added? I keep running into roadblocks trying to get this piece of software to run. It honestly shouldn't be anywhere near so difficult. Disclaimer: I am completely unfamiliar with python and everything related to it, and I imagine most people are (like, in general).

So I installed Python. Just the latest version, because why not. It's important to note that python and its commands should be on the PATH, because otherwise the install command in the docs won't work. Docs doesn't say anything about that, and just blindly assumes the pip command works.

No juice. The installer complains that it only runs on a very specific set of python versions. This is bad, because I now have to go and hunt down the exact right version. And people who *are* familiar with python, might have the wrong version installed. And, after having found the right version of python and trying to install it, the installer complained that another version of python is already installed and must be uninstalled first. That's why it's bad also for people who already work with python - they are likely to be on the latest stable version.

And before you go "that's out of scope" - I don't think it is. It's a requirement for whisper to run, so it's just as important to install python correctly, as it is to install whisper itself. On top of that, it is extremely likely that python hasn't been installed, making it ever more important to get it right, because it's part of the experience of getting this piece of software running.

After that, the pip command started working. I *am* familiar with Node.js, and so I created a directory and cd into it, run pip from there, and expected whisper to install into the current directory. Little did I know, pip doesn't install there. And I haven't the foggiest where it's installed whisper to. And it certainly doesn't put itself on the PATH:

```
c:\Projects\whisper>pip install -U openai-whisper
[success install, so snip]

c:\Projects\whisper>whisper --help
'whisper' is not recognized as an internal or external command,
operable program or batch file.
```

So in the docs, a step is missing where it says where to cd to, in order to run whisper. Because it's not in the directory above.

And then in preparation of actually getting whisper to work, I would expect the docs to list some (or all) formats it supports. Because I have no way of knowing, and the docs doesn't say anything about it.

**Comment by Purfview** [comment]
>  I keep running into roadblocks trying to get this piece of software to run. It honestly shouldn't be anywhere near so difficult. Disclaimer: I am completely unfamiliar with python and everything related to it, and I imagine most people are (like, in general).

Install info is on the main page -> https://github.com/openai/whisper#setup

If Python is too complicated then you can use standalone version  -> https://github.com/Purfview/whisper-standalone-win

> And then in preparation of actually getting whisper to work, I would expect the docs to list some (or all) formats it supports. Because I have no way of knowing, and the docs doesn't say anything about it.

All formats supported by ffmpeg.

**Reply by thany** [reply]
Install info, yes. But not a clear step-by-step. Which is absolutely needed, because this software doesn't come will an installer like most other software. Perhaps it should.

So my goal was not for my particular question to get answered here, and that be it, but for the "install info" to become clear to people unfamiliar with pythin, pip, and whisper.

**Reply by ryanheise** [reply]
> this software doesn't come will an installer like most other software. Perhaps it should.

It's open source software, probably to encourage other developers to build on it, to improve it, rather than for the original developers to do everything. After all, one of the things they've permitted other developers to do by making it open source is for those other developers to make their own easy-to-use installers, just like @Purfview did. Since you wanted something easy that you could just double-click on, and since @Purfview created that and shared the link with you, I'm not quite sure why you didn't appreciate that in your response.

**Reply by thany** [reply]
I do appreciate it. Sorry I didn't realise I had to make that explicit.

But the point is that it shouldn't have to be that way. If one tool exists that is difficult to use, three more will (eventually) pop up to mitigate that, and they'll all do the same thing slightly differently. They will also fall behind on the latest updates (eventually). And the worst is that it becomes a tool-jungle.

What @Purfview did is good work, obviously, but I feel their work should be part of the main repo.

Opensource, frankly, is not an excuse. Not in the last place because most opensource projects provide binaries that can be used as-is, for most popular platforms. But also, there's a concern with trustworthyness. With all due respect to @Purfview, anyone could set up a repo like that, provide an executable for whatever platform they choose to target, and let it perform malicious things. This is probably the most important reason why "first party" or "official" binaries are oftenly preferred.

And they don't have to be binaries, as long as it's not difficult or confusing to install for users unfamiliar with the technology stack.

**Reply by jonathgh** [reply]
We built an installer, called [WhisperScript](https://getwavery.com), which installs in one click on both windows and macOS, because we heard from a lot of people that the process of installing Whisper was a bit more challenging than they expected. We added some quality of life features, like a transcript editor, video player, and speaker separation for good measure on top of the original CLI version.

![WhisperScript-Electron](https://github.com/user-attachments/assets/ae3b97b9-fa3d-409a-86b4-9da5d5864b70)
https://getwavery.com

**Comment by phineas-pta** [comment]
> Install info, yes. But not a clear step-by-step. Which is absolutely needed, because this software doesn't come will an installer like most other software. Perhaps it should.

whisper is not an end-to-end software with installer and such, it's a python package mean to be use in a software, imagine it like a piece of lego to be used in a lego set

> Opensource, frankly, is not an excuse. Not in the last place because most opensource projects provide binaries that can be used as-is, for most popular platforms.

no there're plenty projects like whisper, not everything has installer

so if u not familiar with python, prefer ready-to-use program like what Purfview has done
